tunnel/intercept/tproxy/tproxy_linux.go

```diff
@@ -458,32 +456,36 @@ func (self *tProxy) addInterceptAddr(interceptAddr *intercept.InterceptAddress,
 }
 
 func (self *tProxy) StopIntercepting(tracker intercept.AddressTracker) error {
-	// keep track of routes used by all intercepts. use a map to avoid duplicates
-	routes := map[string]*net.IPNet{}
+	var errors []error
+
 	for _, addr := range self.addresses {
 		err := self.interceptor.ipt.Delete(mangleTable, dstChain, addr.TproxySpec...)
 		if err != nil {
-			return fmt.Errorf("failed to remove iptables rule for service %s: %v", self.service.Name, err)
+			errors = append(errors, err)
+			logrus.Errorf("failed to remove iptables rule for service %s: %v", self.service.Name, err)
 		}
 		if self.interceptor.lanIf != "" {
 			err = self.interceptor.ipt.Delete(filterTable, dstChain, addr.AcceptSpec...)
 			if err != nil {
-				return fmt.Errorf("failed to remove iptables rule for service %s: %v", self.service.Name, err)
+				errors = append(errors, err)
+				logrus.Errorf("failed to remove iptables rule for service %s: %v", self.service.Name, err)
 			}
 		}
-		ipn := addr.IpNet()
-		routes[ipn.String()] = ipn
-	}
```

Comment:
This was doing addresses uniquely, but we had the same address twice, once for UDP and once for TCP. If you only remove once, the address tracker returns false the first time because it knows it's used twice.